gui/wxpython/startup/utils.py

```diff
@@ -18,41 +18,48 @@
 
 import os
 import shutil
+import tempfile
+import sys
 
 
 def get_possible_database_path():
-    """Finds a path to what is possibly a GRASS Database. 
+    """Finds the directory to what is possibly a GRASS Database.
 
     Looks for directory named grassdata in the usual locations.
 
-    Returns the path as a string or None if nothing was found, so the
-    return value can be used to test if the directory was found.
+    Returns the path as a string or None if nothing was found, so the return 
+    value can be used to test if the directory was found.  
     """
-    home = os.path.expanduser('~')
+    home = os.path.expanduser('~')           
+    path = None     
     # try some common directories for grassdata
     # grassdata (lowercase) in home for Linux (first choice)
     # Documents and My Documents for Windows
     # potential translations (old Windows and some Linux)
     # but ~ and ~/Documents should cover most of the cases
-    # ordered by preference and then likelihood
+    # ordered by platform preference      
+            
+    # Search independent "grassdata" directories      
     candidates = [
-        os.path.join(home, "grassdata"),
-        os.path.join(home, "Documents", "grassdata"),
-        os.path.join(home, "My Documents", "grassdata"),
+        os.path.join(home),
+        os.path.join(home, "Documents")
     ]
+    
     try:
         # here goes everything which has potential unicode issues
-        candidates.append(os.path.join(home, _("Documents"), "grassdata"))
-        candidates.append(os.path.join(home, _("My Documents"), "grassdata"))
+        candidates.append(os.path.join(home, _("Documents")))
     except UnicodeDecodeError:
         # just ignore the errors if it doesn't work
         pass
-    path = None
-    for candidate in candidates:
+
+    # Find possible database path
+    for candidate in candidates:         
         if os.path.exists(candidate):
-            path = candidate
-            break  # get the first match
-    return path
+            for subdir in next(os.walk(candidate))[1]:
+                if 'grassdata' in subdir.lower():
+                    path = os.path.join(candidate, subdir)
+                    break                
```

Comment:
break breaks only the inner loop, you need to return the path right away:
```
    for candidate in candidates:
        if os.path.exists(candidate):
            for subdir in next(os.walk(candidate))[1]:
                if 'grassdata' in subdir.lower():
                    return os.path.join(candidate, subdir)
    return None
```

then you don't need to define path variable at all.